docs/build-on-linea/gas-fees.mdx

```diff
@@ -1,139 +1,108 @@
+    over time. Instead, transactions are added to the pending queue until the gas price on the network drops sufficiently for the
+    transactions to be included.
 
-1.  On Linea, the base fee is **set at 7 wei, the minimum fee allowed by the Ethereum protocol after EIP-1559**. With all blocks created by Linea using up to 30M gas (less than 50% of the maximum block size of 61M gas), the fee decreases by 12.5% per block, effectively keeping it at a stable 7 wei.
-2.  Transactions won't be sequenced if the `gasPrice` or `maxPriorityFeePerGas` falls below a certain threshold, which adjusts over time.
+The gas cost to submit your transaction and include it on Ethereum involves the following fee components:
 
-# How do I make sure my transactions go through?
+- Layer 2 fee - The L2 fee (execution fee) is the cost to include your transaction on the Linea sequencer and is calculated
+    using a similar formula to Ethereum. 
+- Layer 1 fee - The L1 fee is the cost to publish your L2 transaction onto Ethereum and can vary based on the size
+    of the transaction calldata.
 
-### MetaMask Users
+You can [use the `linea_estimateGas`](../reference/api/linea-estimategas.mdx) gas API to estimate your gas costs; the result of the
+call provides an estimate that includes the layer 1 and layer 2 fees.
 
-**MetaMask should automatically calculate the necessary fees to ensure your transaction goes through.**
+## Calculate transaction fees
 
-### Non-MetaMask Users
+Use the [`linea_estimateGas`](../reference/api/linea-estimategas.mdx) API to estimate the gas cost for sending a
+transaction. The API will return the recommended gas limit, the base fee per gas, and the priority fee per gas.
 
-:::info
- 
+:::info important
 
-Developers can access the Gas API used by MetaMask through Infura for transactions submitted from the backend! 
-
-Go [here](https://www.infura.io/platform/gas-api) for more details. 
+`linea_estimateGas` uses the same parameters as [`eth_estimateGas`](https://docs.infura.io/api/networks/linea/json-rpc-methods/eth_estimategas),
+however we recommend using `linea_estimateGas` for more accurate results.
 
 :::
 
-To ensure that your transaction gets included by the sequencer, we recommend that non-MetaMask users use EIP-1559 with the following settings:
+The following example executes the `linea_estimateGas` API on the Goerli testnet.
 
-- maxBaseFee = 1.35 \* previousBlockMaxBaseFee (equivalent to the medium ("Market") setting on MetaMask)
+<Tabs className="my-tabs">
+  <TabItem value="curl" label="curl" default>
 
-- `maxPriorityFeePerGas` = reward value from eth_feeHistory( 5 blocks, latest, 20th percentile)
+```bash
+curl -X POST --data '{"jsonrpc": "2.0","method": "linea_estimateGas","params": [{"from": "0xFE3B557E8Fb62b89F4916B721be55cEb828dBd73","value": "0x1","gasPrice":"0x100000000","gas":"0x21000"}],"id": 53}' https://rpc.goerli.linea.build
```

Comment:
We will need to update this URL to mainnet once it's ready. I don't think we need to have both.